src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -395,6 +396,28 @@ trait UpdateSchema {
   val updateSchema: StructType
 }
 
-abstract class DeleteSchema {
-  val deleteSchema: StructType = StructType(Seq(StructField("id", DataTypes.LongType)))
+trait UpdateSchemaFromUpsertSchema extends UpsertSchema with UpdateSchema {
+  override val updateSchema: StructType = upsertSchema
 }
```

Comment:
If I'm not missing anything, UpdateSchema trait isn't used anywhere, so this new one shouldn't be needed too and both can be dropped

Otherwise I'm curious about the diffrerences between update&upsert schemas. By the name it seems that maybe one is keeping unspecified (null?) fields and other resets all fields, but then schema is fixed and always lists all fields? :thinking:

Reply:
oh, I think I can see how update<>upsert may be different, just that currently trait isn't used and all existing update schemas choose to be same as upsert schema

Reply:
The difference is that not all Relation support update schema. There is different handling in SchemaService, where some don't support both upsert and update and some do, which I think can make sense if the relation doesn't support updating?